File: backend/Models/Task.cs
namespace backend.Models;

using System.ComponentModel.DataAnnotations;

public class Task : BaseModel
{
    [MaxLength(60)]
    public string Title { get; set; } = null!;
    [MaxLength(200)]
    public Task? Parent { get; set; }
    public string Description { get; set; } = null!;
    public DateTime DueDate { get; set; } = DateTime.Now.AddDays(7);
    public List List { get; set; } = null!;
    public int ListId { get; set; }
    public PriorityTask Priority { get; set; } = PriorityTask.low;
    public ICollection<User>? Users { get; set; } = null!;
    public User CreatedBy { get; set; } = null!;
    public int CreatedById { get; set; }

    public enum PriorityTask
    {
        low,
        medium,
        high
    }

}

File: frontend/src/models/task.ts
import { emptyUser, User } from './user';

interface TaskInitialState {
  tasks: Task[];
}

export enum Priority {
  low,
  medium,
  high,
}

export interface SubTask {
  id: number;
  title: string;
  done: boolean;
}

export interface Task {
  id: number;
  title: string;
  description: string;
  priorityTask: Priority;
  subTasks: SubTask[];
  comments: Comment[];
  users: User[];
  createdBy: User;
  dueDate: Date;
}

export const initialTaskValue: Task = {
  id: 0,
  title: '',
  description: '',
  priorityTask: Priority.low,
  subTasks: [],
  comments: [],
  users: [],
  createdBy: emptyUser,
  dueDate: new Date(),
};

export const initialTaskState: TaskInitialState = {
  tasks: [],
};
